packages/client/src/object/convertWireToOsdkObjects/InterfaceHolder.ts

```diff
@@ -23,4 +23,6 @@ export interface InterfaceHolder<
   _Q extends Osdk.Instance<any> = never,
 > extends BaseHolder {
   [InterfaceDefRef]: InterfaceMetadata;
+
+  $metadata: InterfaceMetadata;
```

Comment:
I think this will lead to confusion about state since the Object and Interface instance of this are both called `$metadata`. 

```ts
const employee = await loadEa();
employee.$metadata;
employee.$as(FullTimer).$metadata;
```

The pattern we used for `$apiName` was to set `$objectType` to always be the underlying and for `$apiName` to be contextual. 

Maybe we should do the same thing here? 

```
employee.$metadata.apiName == "Employee";
employee.$objectMetadata.apiName == "FullTimer";
employee.$as(FullTimer).$metadata.apiName == "FullTimer";
```